### PR TITLE
Introduce GitHub Actions

### PR DESCRIPTION
**Pull request by t-mat**
I'd like to introduce GitHub Actions.  Since there're known issues, I don't claim it replaces travis-ci and appveyor for now.

Please take a look at the result.  https://github.com/t-mat/xxHash/actions/runs/876834438

# Good points

- Since every GitHub users can use GitHub Actions, everyone run it without other accounts.
- GitHub Actions is highly integrated to GitHub UX.

# Critical issue of this PR

- AVX512 related tests are disabled.
  Because default environment of GitHub Actions doesn't guarantee to support AVX512.
  As of May 2021, they're using Xeon E5-2673 (which doesn't support AVX512) and Xeon Platinum 8171M (which supports AVX512).
  See also https://github.com/actions/runner/issues/1069

# Known issues

- It fails for `make usan`.
  - Currently, it doesn't fail at travis-ci.  Because we're using older version of ubuntu and clang at travis-ci.
  - To understand what's happening, please run `make usan` at your local environment.
  - In short, it actually fails.

- GitHub Actions doesn't support Visual Studio 2015 and 2013.
  https://github.com/actions/virtual-environments/issues/387

- GitHub Actions doesn't have native runner for { ARM, ARM64, PPC64LE, S390X }.

- GitHub Actions doesn't have MSVC 2013 and MSVC 2015.

- I'm still not sure how properly install clang for msys2.

### CODE REVIEW

**Comment by t-mat** [comment]
PR #531 fixes `make usan` issue.

**Comment by hamidnn** [comment]
***@***.***

در تاریخ چهارشنبه ۲۶ مهٔ ۲۰۲۱،‏ ۵:۴۹ Takayuki MATSUOKA <
***@***.***> نوشت:

**Comment by Cyan4973** [comment]
Thanks for taking this over @t-mat .

Moving tests towards Github Action is a good move.
It is expected that not everything can be transferred, as explained in your PR.
But that's nonetheless a good start.

Do you foresee additional actions to be included in this PR,
or is it ready to merge ? (other updates could be done later in a separate PR).

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+#   Also sometimes it reports false positives.
+#
+# - GitHub Actions doesn't support Visual Studio 2015 and 2013.
+#   https://github.com/actions/virtual-environments/issues/387
+#
+# - Setup procedure for msys2 environment is painfully slow.  It takes
+#   3..5 minutes.
+#
+# - Sometimes apt-get fails to retrieve package information.
+#   I have absolutely no idea.
```
Comment (Cyan4973) [inline .github/workflows/ci.yml]: from experience, `apt-get` is not 100% reliable in many CI, including GA.

Comment (t-mat) [reply .github/workflows/ci.yml]: Thanks.  I'll remove this known issue from the code.

**Comment by t-mat** [comment]
> Do you foresee additional actions to be included in this PR,
> or is it ready to merge ? (other updates could be done later in a separate PR).

I think it's ready to merge.

Although I'd like to tweak the test script for AVX512.  But it should be other PR.

**Review by Cyan4973 — APPROVED** [review]
